src/nccl_ofi_ofiutils.cpp

```diff
@@ -219,38 +219,135 @@ int nccl_ofi_ofiutils_get_providers(const char *prov_include,
 }
 
 
-int nccl_ofi_ofiutils_init_connection(struct fi_info *info, struct fid_domain *domain,
-				      struct fid_ep **ep, struct fid_av **av, struct fid_cq *cq)
+/*
+ * @brief	Release libfabric endpoint, address vector, and completion queue
+ */
+void nccl_ofi_ofiutils_ep_release(fid_ep_ptr& ep, fid_av_ptr& av, int dev_id)
+{
+	ep.reset();
+	av.reset();
```

Comment:
Note: I kept explicitly cleaning up the Libfabric resources in this iteration to stay consistent with the previously implementation, but I can probably just remove the explicit call and have the destructor get automatically called when calling `delete` on the owning instance.